### PR TITLE
Hack n learn how to add an optional ArgParser capability to a generated project

### PR DESCRIPTION
**Pull request by aarbizu**

### CODE REVIEW

File: src/main/kotlin/com/pk/Main.kt [hunk 1]
```diff
@@ -65,10 +65,18 @@ fun main(args: Array<String>) = mainBody {
...
         )
 
+        val defaultDependencies = mutableListOf<Dependency>()
+
+        if (useArgParser) {
+            defaultDependencies.add(Dependency("implementation", "com.xenomachina", "kotlin-argparser"))
+        }
+
+        defaultDependencies.addAll(dependencies())
+
```
Comment (aarbizu) [inline src/main/kotlin/com/pk/Main.kt]: Maybe there's a better (more idiomatic?) kotlin way to do this with union() and immutable collections, but I didn't spend too much time trying to figure it out...

**Comment by pallavkothari** [comment]
this looks awesome! it would be cool if this was on by default and the flag just lets you opt out instead of opt in (to optimize for the number of flags I envision typing in regularly)

meanwhile, trying to figure out why CI didn't work..

**Comment by pallavkothari** [comment]
ok, I was missing a magic word -- let's see if CI kicks in on your next edit

File: src/main/kotlin/com/pk/Main.kt [hunk 2]
```diff
@@ -41,33 +42,47 @@ class MyArgs(parser: ArgParser) {
...
+    )
 }
 
 fun main(args: Array<String>) = mainBody {
     ArgParser(
         args,
         helpFormatter = DefaultHelpFormatter(
             epilogue = "Sample usage: " +
-                "ktinit -c --group-id com.pk --artifact-id kotlindemo -d com.google.guava:guava"
+                "ktinit -c --group-id com.pk --artifact-id kotlindemo -d com.google.guava:guava --parseArgs"
```
Comment (pallavkothari) [inline src/main/kotlin/com/pk/Main.kt]: [--no-arg-parsing] where [] denotes optional

File: src/main/kotlin/com/pk/Main.kt [hunk 3]
```diff
@@ -43,9 +43,9 @@ class MyArgs(parser: ArgParser) {
         }
     }.default<List<Dependency>>(listOf())
 
-    val useArgParser by parser.flagging(
-        "-p", "--parseArgs",
-        help = "set up command-line arg parsing"
+    val noArgs by parser.flagging(
+        "-n", "--no-arg-parsing",
```
Comment (pallavkothari) [inline src/main/kotlin/com/pk/Main.kt]: Just FYI another way to do this while keeping your original code would have been to leave the parseArgs flag as is and just default it to true here. That's what I was imagining, but flipping the logic works too.

**Pull request merged**